test/ServiceManagerTest.php

```diff
@@ -14,9 +14,10 @@ class ServiceManagerTest extends AbstractMezzioContainerConfigTest
     use SharedTestTrait;
 
     /**
+     * @param array<string,mixed> $config
      * @return ServiceManager
      */
-    public function createContainer(array $config) : ContainerInterface
+    public function createContainer(array $config): ContainerInterface
     {
         return new ServiceManager($config);
```

Comment:
I think the parameter `array<string, mixed> $config` is a bit too lax here, heh...

Reply:
https://github.com/laminas/laminas-servicemanager/blob/3.11.x/src/ConfigInterface.php

We could the psalm type from laminas-servicemanager.
Would require a version bump here but that should be fine imho.

Reply:
I have tried setting the `@psalm-param` as defined in `ServiceManager`. But this is more specific than the method definition in `AbstractContainerTest`, which is also used by `AuraDiTest`. The configuration object of AuraDi is also not compatible with that of the config array for `ServiceManager`. Not really sure, how to solve this.

Reply:
So `AuraDi` integration test is in this component as well? 

IMHO: kick all integration tests and instead add them to the corresponding components. No integration tests should be needed in this component.
We can, ofc, create a fake integration to have integration tests.
But yes, if we add psalm-types, ofc, anyone has to match these.
So when moving integration tests to their integrations, the test cases have to adapt as well.